compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -5390,6 +5386,25 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         case UNWRAP_FROM_THROWABLE =>
           // js.special.unwrapFromThrowable(arg)
           js.UnwrapFromThrowable(genArgs1)
+
+        case LINKTIME_PROPERTY_BOOLEAN |
+             LINKTIME_PROPERTY_INT |
+             LINKTIME_PROPERTY_STRING =>
+          // LinkingInfo.linkTimePropertyXXX("...")
+          val arg = genArgs1
+          val tpe: jstpe.Type = code match {
```

Comment:
Have you considered using
```suggestion
          val tpe: jstpe.Type = toIRType(tree.tpe) match {
            case jstpe.ClassType(BoxedStringClass, _) => jstpe.StringType
            case iryTpe                               => irType
          }
```
?

Then you don't even need the 3 separate constants. You could use `LINKTIME_PROPERTY` for the 3 of them.